File: DSA_In_CSharp/03_Mathematics/05_GCD.cs
namespace Mathematics
{
    public class GreatestCommonDevisor {
        public int GCD(int a, int b) {
            if (b == 0)
                return a;
            return GCD(b, a % b);
        }
        public void Test() {
            Console.WriteLine(GCD(4, 6));
        }
    }
}

File: DSA_In_Typescript/03_Mathematics/05_GCD.ts
class GreatestCommonDevisor {
    gcd(a: number, b: number): number {
        if (b === 0)
            return a
        return this.gcd(b, a % b)
    }
}
console.log(new GreatestCommonDevisor().gcd(4, 6))